diff --git a/Sources/CCryptoBoringSSL/CMakeLists.txt b/Sources/CCryptoBoringSSL/CMakeLists.txt
index dc7193c..bfc78d2 100644
--- a/Sources/CCryptoBoringSSL/CMakeLists.txt
+++ b/Sources/CCryptoBoringSSL/CMakeLists.txt
@@ -379,6 +379,34 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|Android" AND CMAKE_SYSTEM_PROCESSOR MATC
     crypto/fipsmodule/sha256-armv8.linux.aarch64.S
     crypto/fipsmodule/sha512-armv8.linux.aarch64.S
     crypto/fipsmodule/vpaes-armv8.linux.aarch64.S)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|Android" AND CMAKE_SYSTEM_PROCESSOR MATCHES "i686")
+  target_sources(CCryptoBoringSSL PRIVATE
+    crypto/chacha/chacha-x86.linux.x86.S
+    crypto/fipsmodule/aesni-x86.linux.x86.S
+    crypto/fipsmodule/bn-586.linux.x86.S
+    crypto/fipsmodule/co-586.linux.x86.S
+    crypto/fipsmodule/ghash-x86.linux.x86.S
+    crypto/fipsmodule/ghash-ssse3-x86.linux.x86.S
+    crypto/fipsmodule/md5-586.linux.x86.S
+    crypto/fipsmodule/sha1-586.linux.x86.S
+    crypto/fipsmodule/sha256-586.linux.x86.S
+    crypto/fipsmodule/sha512-586.linux.x86.S
+    crypto/fipsmodule/vpaes-x86.linux.x86.S
+    crypto/fipsmodule/x86-mont.linux.x86.S)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|Android" AND CMAKE_SYSTEM_PROCESSOR MATCHES "armv7-a")
+  target_sources(CCryptoBoringSSL PRIVATE
+    crypto/chacha/chacha-armv4.linux.arm.S
+    crypto/curve25519/asm/x25519-asm-arm.S
+    crypto/fipsmodule/aesv8-armx32.linux.arm.S
+    crypto/fipsmodule/armv4-mont.linux.arm.S
+    crypto/fipsmodule/bsaes-armv7.linux.arm.S
+    crypto/fipsmodule/ghash-armv4.linux.arm.S
+    crypto/fipsmodule/ghashv8-armx32.linux.arm.S
+    crypto/fipsmodule/sha1-armv4-large.linux.arm.S
+    crypto/fipsmodule/sha256-armv4.linux.arm.S
+    crypto/fipsmodule/sha512-armv4.linux.arm.S
+    crypto/fipsmodule/vpaes-armv7.linux.arm.S
+	crypto/poly1305/poly1305_arm_asm.S)
 endif()

 target_include_directories(CCryptoBoringSSL PUBLIC
